### ISSUE TITLE
Freezing issue

### ISSUE THREAD

**Issue by Dimand** [martinbutt/openmiko#13]
**Describe the bug**
Camera freezes randomly. Total time uptime appears random. 

Last frame of camera in frigate
https://imgur.com/a/tNQD88G

SSH also crashes :( Fixed after power cycle.

**Specs**
v1.0.0-alpha.2 on Wyze v2.

**To Reproduce**
Unsure.

**Expected behavior**
No crashes.

NOTE.
I have commented out VIDEO_DEV_3 in openmiko.conf so I could rotate some cameras with videocapture_settings.json
It appears to not work without doing this as /dev/video5 is not defined in the json file. However cameras I haven't flipped have also crashed.

I will try turning off ENABLE_AUDIO and see  if that helps.

openmiko.conf
```
# OpenMiko configuration file

#ENABLE_DEBUG=0

# Starts ffmpeg on startup recording from the v4l2rtspserver
#ENABLE_RECORDING=0

# Start the micropython based api
ENABLE_API=1

# Use a swapfile on the sdcard
# ENABLE_SWAP=1
# Use this setting if swapfile is a dedicated partition
# SWAPFILE=/dev/mmcblk0p1

# Set parition to use for mounting at /sdcard
# SD_PARTITION=/dev/mmcblk0p2
# SD_FILESYSTEM=vfat

# Enable MJPEG over HTTP
ENABLE_MJPEG_OVER_HTTP=1
MJPEG_FPS=15

# Script that detects nighttime/daytime and turns on IR LEDs
ENABLE_AUTONIGHT=1
AUTONIGHT_PARAMS="-j 3 -w 3 -1 1200000 -2 930000,14,10 -3 3000,17,8"

# Set alternate videocapture settings file
VIDEO_CAPTURE_SETTINGS=/etc/videocapture_settings.json

# Set enable audio (is currently buggy, use at your own risk!)
ENABLE_AUDIO=1

# Make sure these settings match the /etc/videocapture_settings.json file
VIDEO_DEV_1=/dev/video3
VIDEO_DEV_2=/dev/video4
#VIDEO_DEV_3=/dev/video5

# Use 8189fs for WyzeCam V2. For the WyzeCam Pan and Dafang use 8189es
WIFI_MODULE=8189fs

# Disables LEDs on the camera
DISABLE_LEDS=0

# Enable logging
ENABLE_LOGGING=0


```

videocapture_settings.json
```{
  "comment": "2 encoders (1920x1080 H264, 1920x1080 MJPEG)",
  "general_settings": {
    "flip_vertical": 1,
    "flip_horizontal": 1,
    "timestamp_24h": 1,
    "timestamp_location": 0,
    "show_timestamp": 1,
    "enable_audio": 1,
    "enable_logging": 0
  },
  "frame_sources": [{
    "id": 0,
    "pic_width": 1920,
    "pic_height": 1080,
    "group": 0,
    "pixel_format": "PIX_FMT_NV12",
    "buffer_size": 2,
    "channel_type": "FS_PHY_CHANNEL",
    "crop_enable": 0,
    "crop_top": 0,
    "crop_left": 0,
    "crop_width": 0,
    "crop_height": 0,
    "scaling_enable": 1,
    "scaling_width": 1920,
    "scaling_height": 1080,
    "frame_rate_numerator": 25,
    "frame_rate_denominator": 1
  }],
  "encoders": [{
    "channel": 0,
    "group": 0,
    "v4l2_device_path": "/dev/video3",
    "payload_type": "PT_H264",
    "buffer_size": 0,
    "profile": 0,
    "mode": "ENC_RC_MODE_H264VBR",
    "frame_rate_numerator": 25,
    "frame_rate_denominator": 1,
    "max_group_of_pictures": 10,
    "max_qp": 38,
    "min_qp": 15,
    "frame_qp_step": 3,
    "gop_qp_step": 15,
    "pic_width": 1920,
    "pic_height": 1080,

    "h264vbr_settings": {
      "statistics_interval": 1,
      "max_bitrate": 900,
      "change_pos": 50
    },
    "h264cbr_settings": {
      "bitrate": 500,
      "max_fps": 100,
      "min_fps": 1,
      "ibiaslvl": 2,
      "adaptive_mode": 0,
      "gop_relation": 0
    }
  },
  {
    "channel": 1,
    "group": 0,
    "v4l2_device_path": "/dev/video4",
    "payload_type": "PT_JPEG",
    "buffer_size": 0,
    "profile": 0,
    "mode": "ENC_RC_MODE_H264VBR",
    "frame_rate_numerator": 25,
    "frame_rate_denominator": 1,
    "max_group_of_pictures": 10,
    "max_qp": 38,
    "min_qp": 15,
    "frame_qp_step": 3,
    "gop_qp_step": 15,
    "pic_width": 1920,
    "pic_height": 1080,

    "h264vbr_settings": {
      "statistics_interval": 1,
      "max_bitrate": 500,
      "change_pos": 50
    },
    "h264cbr_settings": {
      "bitrate": 500,
      "max_fps": 100,
      "min_fps": 1,
      "ibiaslvl": 2,
      "adaptive_mode": 0,
      "gop_relation": 0
    }
  }],
  "bindings": [
    {
      "note": "Bind framesource 0,0 to OSD 0,0 ",
      "source": {
        "device": "DEV_ID_FS",
        "group": 0,
        "output": 0
      },
      "target": {
        "device": "DEV_ID_OSD",
        "group": 0,
        "output": 0
      }
    },
    {
      "note": "Bind OSD 0,0 to encoder 0,0",
      "source": {
        "device": "DEV_ID_FS",
        "group": 0,
        "output": 0
      },
      "target": {
        "device": "DEV_ID_ENC",
        "group": 0,
        "output": 0
      }
    }
  ]
}
```

**Comment by martinbutt** [comment]
I'll try to replicate this using your config this weekend.

**Comment by Dimand** [comment]
Some updates.

Had another one freeze in the afternoon.
https://imgur.com/a/Qf21ZZm

This one is outside and I noticed that the IR lights had still come on even though it froze during daylight (only getting to it tonight). I shone my torch at it for ~20 seconds and the IR switched off, then switched on again when I took the torch off it.
I think this confirms that the camera + processor are still working and the orange light is still flashing as well.
This one had ENABLE_AUDIO=0 in the config (with reboot) as of this morning.

Now for the interesting part. This came good without me power cycling the camera. I did two things, firstly, the test with the torch and the IR illumination. But I don't think it was that as the timing is not quite right. I also had a weird bug with home assistant where I could not access it through my hostname address but it worked fine through a direct IP connection. I'm pretty sure the camera came online when I restarted the home assistant instance to solve this issue. SSH access restored and all, no issue with accessing HA.

Frigate is running on a docker container on my truenas scale box. I have the frigate integration and cards utilised in home assistant. Home assistant runs on a separate machine (Pi4).
https://github.com/blakeblackshear/frigate-hass-integration
https://github.com/dermotduffy/frigate-hass-card

I also have a node red automation that pulls snapshots with the http://ip:8080/?action=snapshot link. I did this 15 seconds before the camera froze based on the timestamp. So I am thinking there is some issue with the http requests and home assistant? I'll wait till another one freezes and have a play.
